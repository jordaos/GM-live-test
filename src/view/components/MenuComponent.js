import React from 'react';

const MenuComponent = ({ menuList }) => {
    return (
        <div className="col-sm-3 col-md-2 sidebar">
            <ul className="nav nav-sidebar">
                <li>
                    <div className="media">
                        <div className="media-left">
                            <img className="media-object" src="data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiIHN0YW5kYWxvbmU9InllcyI/PjxzdmcgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB3aWR0aD0iNjQiIGhlaWdodD0iNjQiIHZpZXdCb3g9IjAgMCA2NCA2NCIgcHJlc2VydmVBc3BlY3RSYXRpbz0ibm9uZSI+PGRlZnMvPjxyZWN0IHdpZHRoPSI2NCIgaGVpZ2h0PSI2NCIgZmlsbD0iI0VFRUVFRSIvPjxnPjx0ZXh0IHg9IjEzLjQ2MDkzNzUiIHk9IjMyIiBzdHlsZT0iZmlsbDojQUFBQUFBO2ZvbnQtd2VpZ2h0OmJvbGQ7Zm9udC1mYW1pbHk6QXJpYWwsIEhlbHZldGljYSwgT3BlbiBTYW5zLCBzYW5zLXNlcmlmLCBtb25vc3BhY2U7Zm9udC1zaXplOjEwcHQ7ZG9taW5hbnQtYmFzZWxpbmU6Y2VudHJhbCI+NjR4NjQ8L3RleHQ+PC9nPjwvc3ZnPg==" alt="..." />
                        </div>
                        <div className="media-body">
                            <h4 className="media-heading">Jord√£o Macedo</h4>
                            @jordaos
                        </div>
                    </div>
                </li>
                {menuList.map((item, index) => {
                    return (<li key={index}><a href={item.href} className={item.active ? 'active' : ''}>{item.title}</a></li>)
                })}
            </ul>
        </div>
    );
}
export default MenuComponent;