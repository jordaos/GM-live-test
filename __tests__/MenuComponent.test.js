import React from 'react';
import { shallow } from 'enzyme';
import { expect as chaiExpect } from 'chai';
import * as objects from './objects'

import MenuComponent from './../src/view/components/MenuComponent'

describe('Testing MenuComponent', () => {
    it('Render Component', () => {
        const wrapper = shallow(<MenuComponent user={objects.user} menuList={objects.menuList}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('Show two items in menu', () => {
        const wrapper = shallow(<MenuComponent user={objects.user} menuList={objects.menuList}/>);
        chaiExpect(wrapper.find('div.col-sm-3.col-md-2.sidebar > ul > li > a')).to.have.lengthOf(2);
    });

    it('Show correct user informations', () => {
        const wrapper = shallow(<MenuComponent user={objects.user} menuList={objects.menuList}/>);
        chaiExpect(wrapper.find(".media-heading").text()).to.equal("Jord√£o");
    })
});