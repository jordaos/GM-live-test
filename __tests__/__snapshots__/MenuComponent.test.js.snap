// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MenuComponent Render Component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuComponent
    menuList={
      Array [
        Object {
          "active": true,
          "href": "/item1",
          "title": "Item 1",
        },
        Object {
          "active": false,
          "href": "/item2",
          "title": "Item 2",
        },
      ]
    }
    user={
      Object {
        "avatar_url": "https://avatars0.githubusercontent.com/u/7025768?v=4",
        "login": "jordaos",
        "name": "Jordão",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="nav nav-sidebar"
      >
        <li>
          <div
            className="media"
          >
            <div
              className="media-left"
            >
              <img
                alt="User"
                className="media-object img-profile"
                src="https://avatars0.githubusercontent.com/u/7025768?v=4"
              />
            </div>
            <div
              className="media-body"
            >
              <h4
                className="media-heading"
              >
                Jordão
              </h4>
              @
              jordaos
            </div>
          </div>
        </li>
        <li>
          <a
            className="active"
            href="/item1"
          >
            Item 1
          </a>
        </li>
        <li>
          <a
            className=""
            href="/item2"
          >
            Item 2
          </a>
        </li>
      </ul>,
      "className": "col-sm-3 col-md-2 sidebar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <div
              className="media"
            >
              <div
                className="media-left"
              >
                <img
                  alt="User"
                  className="media-object img-profile"
                  src="https://avatars0.githubusercontent.com/u/7025768?v=4"
                />
              </div>
              <div
                className="media-body"
              >
                <h4
                  className="media-heading"
                >
                  Jordão
                </h4>
                @
                jordaos
              </div>
            </div>
          </li>,
          Array [
            <li>
              <a
                className="active"
                href="/item1"
              >
                Item 1
              </a>
            </li>,
            <li>
              <a
                className=""
                href="/item2"
              >
                Item 2
              </a>
            </li>,
          ],
        ],
        "className": "nav nav-sidebar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="media"
            >
              <div
                className="media-left"
              >
                <img
                  alt="User"
                  className="media-object img-profile"
                  src="https://avatars0.githubusercontent.com/u/7025768?v=4"
                />
              </div>
              <div
                className="media-body"
              >
                <h4
                  className="media-heading"
                >
                  Jordão
                </h4>
                @
                jordaos
              </div>
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="media-left"
                >
                  <img
                    alt="User"
                    className="media-object img-profile"
                    src="https://avatars0.githubusercontent.com/u/7025768?v=4"
                  />
                </div>,
                <div
                  className="media-body"
                >
                  <h4
                    className="media-heading"
                  >
                    Jordão
                  </h4>
                  @
                  jordaos
                </div>,
              ],
              "className": "media",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="User"
                    className="media-object img-profile"
                    src="https://avatars0.githubusercontent.com/u/7025768?v=4"
                  />,
                  "className": "media-left",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "User",
                    "className": "media-object img-profile",
                    "src": "https://avatars0.githubusercontent.com/u/7025768?v=4",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h4
                      className="media-heading"
                    >
                      Jordão
                    </h4>,
                    "@",
                    "jordaos",
                  ],
                  "className": "media-body",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Jordão",
                      "className": "media-heading",
                    },
                    "ref": null,
                    "rendered": "Jordão",
                    "type": "h4",
                  },
                  "@",
                  "jordaos",
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="active"
              href="/item1"
            >
              Item 1
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Item 1",
              "className": "active",
              "href": "/item1",
            },
            "ref": null,
            "rendered": "Item 1",
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <a
              className=""
              href="/item2"
            >
              Item 2
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Item 2",
              "className": "",
              "href": "/item2",
            },
            "ref": null,
            "rendered": "Item 2",
            "type": "a",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="nav nav-sidebar"
        >
          <li>
            <div
              className="media"
            >
              <div
                className="media-left"
              >
                <img
                  alt="User"
                  className="media-object img-profile"
                  src="https://avatars0.githubusercontent.com/u/7025768?v=4"
                />
              </div>
              <div
                className="media-body"
              >
                <h4
                  className="media-heading"
                >
                  Jordão
                </h4>
                @
                jordaos
              </div>
            </div>
          </li>
          <li>
            <a
              className="active"
              href="/item1"
            >
              Item 1
            </a>
          </li>
          <li>
            <a
              className=""
              href="/item2"
            >
              Item 2
            </a>
          </li>
        </ul>,
        "className": "col-sm-3 col-md-2 sidebar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <div
                className="media"
              >
                <div
                  className="media-left"
                >
                  <img
                    alt="User"
                    className="media-object img-profile"
                    src="https://avatars0.githubusercontent.com/u/7025768?v=4"
                  />
                </div>
                <div
                  className="media-body"
                >
                  <h4
                    className="media-heading"
                  >
                    Jordão
                  </h4>
                  @
                  jordaos
                </div>
              </div>
            </li>,
            Array [
              <li>
                <a
                  className="active"
                  href="/item1"
                >
                  Item 1
                </a>
              </li>,
              <li>
                <a
                  className=""
                  href="/item2"
                >
                  Item 2
                </a>
              </li>,
            ],
          ],
          "className": "nav nav-sidebar",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="media"
              >
                <div
                  className="media-left"
                >
                  <img
                    alt="User"
                    className="media-object img-profile"
                    src="https://avatars0.githubusercontent.com/u/7025768?v=4"
                  />
                </div>
                <div
                  className="media-body"
                >
                  <h4
                    className="media-heading"
                  >
                    Jordão
                  </h4>
                  @
                  jordaos
                </div>
              </div>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="media-left"
                  >
                    <img
                      alt="User"
                      className="media-object img-profile"
                      src="https://avatars0.githubusercontent.com/u/7025768?v=4"
                    />
                  </div>,
                  <div
                    className="media-body"
                  >
                    <h4
                      className="media-heading"
                    >
                      Jordão
                    </h4>
                    @
                    jordaos
                  </div>,
                ],
                "className": "media",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      alt="User"
                      className="media-object img-profile"
                      src="https://avatars0.githubusercontent.com/u/7025768?v=4"
                    />,
                    "className": "media-left",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "User",
                      "className": "media-object img-profile",
                      "src": "https://avatars0.githubusercontent.com/u/7025768?v=4",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <h4
                        className="media-heading"
                      >
                        Jordão
                      </h4>,
                      "@",
                      "jordaos",
                    ],
                    "className": "media-body",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Jordão",
                        "className": "media-heading",
                      },
                      "ref": null,
                      "rendered": "Jordão",
                      "type": "h4",
                    },
                    "@",
                    "jordaos",
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="active"
                href="/item1"
              >
                Item 1
              </a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Item 1",
                "className": "active",
                "href": "/item1",
              },
              "ref": null,
              "rendered": "Item 1",
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": <a
                className=""
                href="/item2"
              >
                Item 2
              </a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Item 2",
                "className": "",
                "href": "/item2",
              },
              "ref": null,
              "rendered": "Item 2",
              "type": "a",
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
